Adapt CCDBApi to work with MemoryResources

ktf opened the pull request.

ktf committed:
Adapt CCDBApi to work with MemoryResources

ktf commented:
WIP do not merge.

shahor02 commented:
Looks fine, but if you are not going to use the CCDBManagerInstance, then better either leave it unchanged (if somebody will want to get the BLOB in my old definition) or to revert it completely to pre-https://github.com/AliceO2Group/AliceO2/pull/7900 state. In your version you leave the cache wrapper but destroy the cached blob, so it will crash at the next query.

ktf commented:
This seems to work with my prototype CCDB fetcher, which indeed does not need a CCDB manager at this point, so I kept the changes there to the bare minimum. Notice I had to introduce one copy to avoid duplicating the loadFileToMemory to be used with `o2::vector`.

ktf committed:
Do not pollute toplevel o2 namespace with vector

It causes troubles whenever using o2 and std at the same time.
It's probably good enough that we have o2::pmr::vector so that
we can differenciate the purpose without asking people to
avoid "using namespace std".
